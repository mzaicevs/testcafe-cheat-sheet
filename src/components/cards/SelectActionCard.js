import React from "react";
import { actions_select } from "../../snippets";
import MultiCodeCard from "../../base_components/MultiCodeCard";

const SelectActionCard = () => (
  <MultiCodeCard
    cardTitle={actions_select.title}
    codeSections={actions_select.codeSections}
  />
);

export default SelectActionCard;
