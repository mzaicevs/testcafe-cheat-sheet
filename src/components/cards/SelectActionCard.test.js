import React from "react";
import { shallow } from "enzyme";
import SelectActionCard from "./SelectActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("FixtureCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<SelectActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
