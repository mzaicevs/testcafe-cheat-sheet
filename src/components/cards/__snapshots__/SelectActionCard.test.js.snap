// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixtureCard matches snapshot 1`] = `
<CodeCard
  cardTitle={
    <CardLink
      href="https://devexpress.github.io/testcafe/documentation/test-api/actions/select-text.html"
      text="Select Action"
    />
  }
  codeSections={
    Array [
      Object {
        "code": Array [
          "t.selectText( selector [, startPos] [, endPos] [, options] )",
        ],
        "description": "Selects text in input elements of various types",
        "example": "import { ClientFunction, Selector } from 'testcafe';

      fixture \`My fixture\`
          .page \`http://devexpress.github.io/testcafe/example/\`;
      
      const developerNameInput = Selector('#developer-name');
      
      const getElementSelectionStart = ClientFunction(selector => selector().selectionStart);
      const getElementSelectionEnd   = ClientFunction(selector => selector().selectionEnd);
      
      test('Select text within input', async t => {
          await t
              .typeText(developerNameInput, 'Test Cafe', { caretPos: 0 })
              .selectText(developerNameInput, 7, 1);
      
          await t
              .expect(await getElementSelectionStart(developerNameInput)).eql(1)
              .expect(await getElementSelectionEnd(developerNameInput)).eql(7);
      });",
        "language": "javascript",
        "options": "{ speed: Number }",
      },
      Object {
        "code": Array [
          "t.selectTextAreaContent( selector [, startLine] [, startPos] [, endLine] [, endPos] [, options] )",
        ],
        "description": "Select text within a <textarea> element",
        "example": "import { ClientFunction, Selector } from 'testcafe';

      fixture \`My fixture\`
          .page \`http://devexpress.github.io/testcafe/example/\`;
      
      const commentTextArea = Selector('#comments');
      
      const getElementSelectionStart = ClientFunction(selector => selector().selectionStart);
      const getElementSelectionEnd   = ClientFunction(selector => selector().selectionEnd);
      
      test('Select text within textarea', async t => {
          await t
              .click('#tried-test-cafe')
              .typeText(commentTextArea, [
                  'Lorem ipsum dolor sit amet',
                  'consectetur adipiscing elit',
                  'sed do eiusmod tempor'
              ].join(',
'))
              .selectTextAreaContent(commentTextArea, 0, 5, 2, 10);
      
          await t
              .expect(await getElementSelectionStart(commentTextArea)).eql(5)
              .expect(await getElementSelectionEnd(commentTextArea)).eql(67);
      });",
        "language": "javascript",
        "options": "{ speed: Number }",
      },
      Object {
        "code": Array [
          "t.selectEditableContent( startSelector, endSelector [, options] )",
        ],
        "description": "Select HTML elements that have the contentEditable attribute enabled.",
        "example": "import { Selector } from 'testcafe';

      fixture \`My fixture\`
          .page \`http://www.example.com/\`;
      
      test('Delete text within a contentEditable element', async t => {
          await t
              .selectEditableContent('#foreword', '#chapter-3')
              .pressKey('delete')
              .expect(Selector('#chapter-2').exists).notOk()
              .expect(Selector('#chapter-4').exists).ok();
      });",
        "language": "javascript",
        "options": "{ speed: Number }",
      },
    ]
  }
/>
`;
